src/frontend/js/components/Icon/index.tsx

```diff
@@ -9,6 +11,8 @@ interface IconProps {
    */
   title?: string;
   className?: string;
+  onClick?: MouseEventHandler;
```

Comment:
Same idea as my comment on the header above; we should avoid allowing any HTML element to have `onClick` listeners as they are mouse/touch specific and prevent keyboard users, screen reader users to interact.

The idea here would be to have a button that wraps the icon, and let the Icon component only do the job of rendering the icon :)